template/template.go

```diff
@@ -49,140 +41,33 @@ func FromGlobs(paths ...string) (*Template, error) {
 		}
 	}
 
-	t.funcs(DefaultFuncs)
 	return t, nil
 }
 
-type FuncMap map[string]interface{}
-
-func (t *Template) funcs(fm FuncMap) *Template {
-	t.text.Funcs(text_tmpl.FuncMap(fm))
-	t.html.Funcs(html_tmpl.FuncMap(fm))
-	return t
-}
-
-func (t *Template) ExecuteTextString(name string, data interface{}) (string, error) {
+func (t *Template) ExecuteTextString(text string, data interface{}) (string, error) {
+	tmpl, err := t.text.Clone()
+	if err != nil {
+		return "", err
+	}
+	tmpl, err = tmpl.New("").Parse(text)
+	if err != nil {
+		return "", err
+	}
 	var buf bytes.Buffer
-	err := t.ExecuteText(&buf, name, data)
+	err = tmpl.Execute(&buf, data)
 	return buf.String(), err
 }
 
-func (t *Template) ExecuteText(w io.Writer, name string, data interface{}) error {
-	return t.text.ExecuteTemplate(w, name, data)
-}
-
-func (t *Template) ExecuteHTMLString(name string, data interface{}) (string, error) {
+func (t *Template) ExecuteHTMLString(html string, data interface{}) (string, error) {
```

Comment:
You'll need to send a SIGHUP anyway as they're coming from the config. As mentioned on IRC this is just a first pass, ultimately they should be compiled at the config load stage both for efficiency and to catch syntax errors earlier.